### PR TITLE
Implement success? and failure? methods on response

### PR DESCRIPTION
**Pull request by leviferreira**

### CODE REVIEW

File: lib/intelipost/mash.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Intelipost
+  class Mash < ::Hashie::Mash
+    def success?
+      self.status == 'OK'
```
Comment (esampaio) [inline lib/intelipost/mash.rb]: I think you should check if self.status exists first.

File: lib/intelipost/mash.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    end
+
+    def failure?
+      !success?
+    end
+
+    def all_messages
+      self.messages.map(&:text).join(';')
```
Comment (esampaio) [inline lib/intelipost/mash.rb:13]: Same on messages.

File: spec/lib/intelipost/client_spec.rb [hunk 3]
```diff
@@ -84,6 +88,13 @@
       expect(subject.quote.create(volumes)).to have_key(:content)
     end
 
+    it '.quote.create.failure' do
+      response = subject.quote.create({'failure'=>'failure'})
+      expect(response.success?).to eq false
+      expect(response.failure?).to eq true
+      expect(response.all_messages).not_to eq ''
```
Comment (esampaio) [inline spec/lib/intelipost/client_spec.rb:96]: Shouldn't this test that there are error messages?

File: lib/intelipost/client.rb [hunk 4]
```diff
@@ -24,7 +26,7 @@ def connection
       Faraday.new(url: @uri.to_s) do |conn|
         conn.request :json
 
-        conn.response :mashify
+        conn.response :mashify, { mash_class: Intelipost::Mash }
```
Comment (esampaio) [inline lib/intelipost/client.rb:29]: Great solution!

**Comment by leviferreira** [comment]
@esampaio please CR it again

File: lib/intelipost/mash.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+module Intelipost
+  class Mash < ::Hashie::Mash
+    def success?
+      return false unless self.has_key?('status')
+      self.status == 'OK'
```
Comment (esampaio) [inline lib/intelipost/mash.rb]: can be replaced by:

``` ruby
self.has_key? 'status' and a.status == 'OK'
```

File: lib/intelipost/mash.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    end
+
+    def failure?
+      !success?
+    end
+
+    def all_messages
+      return false unless self.has_key?('messages')
```
Comment (esampaio) [inline lib/intelipost/mash.rb]: maybe `nil` instead of `false`?

**Comment by leviferreira** [comment]
@esampaio take another look plz

**Comment by esampaio** [comment]
Awesome, now you just need to bump the gem's version on `lib/intelipost/version.rb` and we can merge it.

**Comment by leviferreira** [comment]
Done :) @esampaio

**Pull request merged**